src/scm/gitlab.js

```diff
@@ -0,0 +1,278 @@
+      })
+      .catch((err) => {
+        showAlert('Failed to push', LEVEL_ERROR);
+      });
+  }
+
+  getAllBranches() {
+    context.repo.id = context.repo.id || this.namesToIds.repos[context.repo.fullName];
+    return getAllItems(Promise.resolve(
+      {
+        token: this.accessToken,
+        items: [],
+        url: `${this.baseUrl}/projects/${context.repo.id}/repository/branches?${this.tokenParam}`
+      }),
+      this.followPaginate,
+      'gitlab'
+    );
+  }
+
+  getCode() {
+    return new Promise((resolve, reject) => {
+      return $.getJSON(
+        `${this.baseUrl}/projects/${context.repo.id}/repository/tree?ref=${context.branch}&recursive=true&${this.tokenParam}`, {}
+      )
+        .then(resolve)
+        .fail(reject)
+    })
+        .then(response => {
+          const re = new RegExp(`(\\${context.config.filetype}|\\.html${context.config.manifestEnabled ? '|^appsscript.json' : ''})$`);
+          const promises = response.filter((tree) => {
+            return tree.type === 'blob' && re.test(tree.path);
+          })
+            .map(tree => {
+              var xx = `${this.baseUrl}/projects/${context.repo.id}/repository/files/${encodeURIComponent(tree.path)}?ref=${context.branch}&${this.tokenParam}`;
+              return new Promise((resolve, reject) => {
+                $.getJSON(xx, {})
+                  .then((content) => {
+                    resolve({file: tree.path, content: decodeURIComponent(escape(atob(content.content)))});
+                  })
+                  .fail(reject)
+              });
+            });
+          return Promise.all(promises);
+        });
+  }
+
+  getNamespaces() {
+    let testUrl = `${this.baseUrl}/groups?${this.tokenParam}`;
+    return getAllItems(Promise.resolve(
+      {
+        token: this.accessToken,
+        items: [],
+        url: testUrl
+      }),
+      this.followPaginate,
+      'gitlab'
+    )
+      .then(groups => {
+        this.namespaces = [this.user].concat(groups.map(group => group.name));
+        return this.namespaces;
+      })
+      .catch((err) => {
+        showAlert('Failed to get user info.', LEVEL_ERROR);
+      });
+  }
+
+  getRepos() { // Named Projects in gitlab
+    return getAllItems(Promise.resolve(
+      {
+        token: this.accessToken,
+        items: [],
+        url: `${this.baseUrl}/users/${this.user}/projects?${this.tokenParam}`
+      }),
+      this.followPaginate,
+      'gitlab'
+    )
+      .then(response => {
+        this.namesToIds.repos = response.reduce((obj, item) => (obj[item.name] = item.id, obj), {});
+        const repos = Object.keys(this.namesToIds.repos);
+        //if current bind still existed, use it
+        const repo = context.bindRepo[context.id];
+        if (repo && $.inArray(repo.fullName, repos) >= 0) {
+          context.repo = repo;
+        }
+        return repos;
+      });
+  }
+
+  createRepo() {
+    const owner = $('#new-repo-owner').val();
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/leonhartX/gas-github/blob/5becb274ac17be98fee6e945e23cbab998e9d302/src/scm/gitlab.js#L184"></span>‘owner’ is assigned a value but never used.</td>
    </tr>
  </tbody>
</table>